### PR TITLE
Add Logger Implementation

### LINKED ISSUE #505: API Compatibility with Ruby Logger
See the Ruby Logger [docs](http://ruby-doc.org/stdlib-2.3.0/libdoc/logger/rdoc/Logger.html) for the API.

### PR DESCRIPTION
**Pull request by blowmage**
Add Logger implementation that is API compatible with ruby's standard library Logger.

```ruby
require "gcloud"

gcloud = Gcloud.new
logging = gcloud.logging

resource = logging.resource "gae_app",
                            module_id: "1",
                            version_id: "20150925t173233"

logger = logging.logger "syslog", resource, env: :production
```

This will close #505.

### CODE REVIEW

File: lib/gcloud/logging/logger.rb [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+      ##
+      # @private The Google Cloud resource to write the log entry with.
+      attr_reader :resource
+
+      ##
+      # @private The Google Cloud labels to write the log entry with.
+      attr_reader :labels
+
+      ##
+      # @private Creates a new Connection instance.
```
Comment (quartzmo) [inline lib/gcloud/logging/logger.rb]: Connection -> Logger

File: lib/gcloud/logging/logger.rb [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+          add 5, nil, message, &block
+        else
+          add 5, message, nil, &block
+        end
+      end
+
+      ##
+      # Log a message if the given severity is high enough. This is the generic
+      # logging method. Users will be more inclined to use {#debug}, {#info},
+      # {#warn}, {#error}, and {#fatal}.
```
Comment (quartzmo) [inline lib/gcloud/logging/logger.rb]: Add @param docs here?

File: lib/gcloud/logging/project.rb [hunk 3]
```diff
@@ -200,6 +201,38 @@ def write_entries entries, log_name: nil, resource: nil, labels: nil
       end
 
       ##
+      # Lists log entries. Use this method to retrieve log entries from Cloud
+      # Logging.
```
Comment (quartzmo) [inline lib/gcloud/logging/project.rb]: Description above was copy/pasted?

File: lib/gcloud/logging/project.rb [hunk 4]
```diff
@@ -200,6 +201,38 @@ def write_entries entries, log_name: nil, resource: nil, labels: nil
...
+      #
+      # @param [Gcloud::Logging::Entry, Array] entries One or more entry objects
```
Comment (quartzmo) [inline lib/gcloud/logging/project.rb]: I don't see `entries` in the method signature.

**Comment by quartzmo** [comment]
Just some small things in the docs (above), otherwise LGTM.

File: lib/gcloud/logging/logger.rb (hunk 1 above)
Comment (blowmage) [inline lib/gcloud/logging/logger.rb]: I wish I had a facepalm emoji...

File: lib/gcloud/logging/logger.rb (hunk 2 above)
Comment (blowmage) [inline lib/gcloud/logging/logger.rb]: Yes, but I think we need to revisit all the docs so I might leave that for when we take a more comprehensive pass on the documentation.

File: lib/gcloud/logging/project.rb (hunk 3 above)
Comment (blowmage) [inline lib/gcloud/logging/project.rb]: Totes.

**Pull request merged**